### ISSUE TITLE
Issues with multiprocessing

### ISSUE THREAD

**Issue by twsl** [smarie/python-pytest-cases#242]
I am having issues with multiprocessing triggering this error message:
https://github.com/smarie/python-pytest-cases/blob/4976c0073a2fad5fbe5de34a5d1199efda0b7da9/src/pytest_cases/common_pytest.py#L647

Here is a minimal example to recreate the issue.
```python

from pytest_cases import parametrize_with_cases
from torch.multiprocessing import Pool, set_start_method

from functools import partial


class TestCases:
    def case_1(self):
        return 2, 4

    def case_2(self):
        return 3, 9


def f(a, b, t):
    if t:
        assert a ** 2 == b
    else:
        assert not (a ** 2 == b)


@parametrize_with_cases("x,y", cases=TestCases)
def test_f_xy(x, y):

    # set_start_method("spawn")
    pool = Pool(processes=2)
    pool.starmap(partial(f), [(x, y, False), (x, y, True)])


```

Is multiprocessing not supported?

**Comment by twsl** [comment]
Added an example to recreate the issue. When running the test it just keep retrying.

**Comment by smarie** [comment]
Thanks @twsl ! Multiprocessing should definitely supported, if it is supported by pytest.

As a workaround I guess that you can put the multiprocessing code in a separate module (for example the conftest.py or another file in the tests folder). That should prevent `pytest-cases` to be loaded several times in parallel.

I'll have a look one of these days

**Comment by smarie** [comment]
I was able to reproduce the issue.
First, we can check that it works with pytest:

```python
from multiprocessing import Pool

import pytest
from functools import partial

def f(a, b, t):
    if t:
        assert a ** 2 == b
    else:
        assert not (a ** 2 == b)

@pytest.mark.parametrize("x,y", [(2, 4), (3, 9)])
def test_f_xy(x, y):
    pool = Pool(processes=2)
    pool.starmap(partial(f), [(x, y, False), (x, y, True)])
```

Running it yields to the two tests failing as expected, and it completes correctly.

While the following hangs as you point out:

```python
from pytest_cases import parametrize_with_cases
from multiprocessing import Pool

from functools import partial


class TestCases:
    def case_1(self):
        return 2, 4

    def case_2(self):
        return 3, 9

def f(a, b, t):
    if t:
        assert a ** 2 == b
    else:
        assert not (a ** 2 == b)

@parametrize_with_cases("x,y", cases=TestCases)
def test_f_xy(x, y):
    pool = Pool(processes=2)
    pool.starmap(partial(f), [(x, y, False), (x, y, True)])
```

**Comment by smarie** [comment]
Apparently we can detect if a process is the main process using this:

```python
try:
    from multiprocessing import parent_process
except ImportError:
    from multiprocessing import current_process
    from multiprocessing.process import _MainProcess

    def is_main_process():
        return isinstance(current_process(), _MainProcess)
else:
    def is_main_process():
        return parent_process() is None
```

We could therefore use this to only raise the `ValueError` if `not is_main_process`.
See https://stackoverflow.com/questions/30790660/python-multiprocessing-name-of-the-main-process

However if we spawn a single process like below, then `is_main_process` seems to return `True`

```python
p = Process(target=partial(f), args=(x, y, True))
p.start()
p.join()
```

So this workaround is not satisfaying. I will therefore simply comment the error-raising line, hoping that it will not make harder to debug a future situation where the config is None.

**Comment by smarie** [comment]
Should be fixed in 3.6.8. let me know !
